3-UsingAPIs/Week1/assignment/ex1-johnWho.js

```diff
@@ -8,22 +8,25 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
+
+const getAnonName = (firstName) => {
+   return new Promise((resolve, reject) => {
   setTimeout(() => {
     if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
+      reject(new Error("You didn't pass in a first name!"));
       return;
     }
 
     const fullName = `${firstName} Doe`;
 
-    callback(fullName);
+    resolve(fullName);
   }, 1000);
-};
-
+});
+}
 function main() {
-  getAnonName('John', console.log);
+  getAnonName('John')
+  .then((fullName) => console.log(fullName))
+  .catch((error) => console.error(error));
```

Comment:
Clean indentation and easy to read! Well done! :raised_hands:  It is also possible to just pass the function without wrapping it in another function

```suggestion
  getAnonName('John')
  .then(console.log)
  .catch(console.error);
```